src/sdam/monitor.ts

```diff
@@ -310,6 +319,7 @@ function checkServer(monitor: Monitor, callback: Callback<Document | null>) {
     }
 
     if (conn) {
+      // TODO: Check if this is correct
```

Comment:
I think what we should probably do here is rename the property on the connection to `isStreamingMonitoringConnection` and then set the value here to `monitor.heartbeatProtocol === 'streaming`.

Reply:
If I'm understanding correctly, at this point we are carrying out our initial connection, so we wouldn't know whether or not the server is capable of streaming yet, right? In that case we should check the server hello, use that to determine whether or not we should be polling or streaming and then set `isStreamingMonitoringConnection` appropriately.